inline-caching/README.md

```diff
@@ -0,0 +1,247 @@
+```bash
+./<location of gemfire>/bin/gfsh
+start locator --name locator
+start server --name='server' --classpath="<project root directory>/cache-loader/build/libs/*" --J="-Dpostgres.username=myuser" --J="-Dpostgres.password=mypass"
+create async-event-queue --listener=io.vmware.event.ItemAsyncEventListener --id=item-writebehind-queue --batch-size=10 --batch-time-interval="20"
+create region --name=item --type=REPLICATE --cache-loader=io.vmware.event.ItemCacheLoader --async-event-queue-id=item-writebehind-queue
+```
+
+Notice that the jars are passed into the classpath for our JDBC Postgres driver and the jar that contains our CacheLoader and AsyncEventListener implementations.
+These classes will be used by GemFire whenever we interact with our newly created `item` region. Our new region also utilizes the options `--cache-loader` and `--async-event-queue-id`. 
+These point to the implementation of the AsyncEventListener and CacheLoader that were referenced earlier.
+
+## Define the data model using Spring Data
+
+The Spring Data model that will be used to represent data in GemFire typically involves defining a set of Java classes that represent the Spring Data model entities.
+For our example, we are just going to use a simple String, but any Java object can be used. 
+
+Here is an example for how to define a Spring Data model using GemFire's repositories that is mapped to our `item` region that we created above
+
+```java
+@Region("/item")
+public interface ItemRepository extends GemfireRepository<String, String> {
+}
+```
+
+Again, we are using Java Strings here for simplicity, but we can use any Java object to store data in GemFire. 
+
+## Web Project
+
+Our `web` project consists of 4 objects:
+1. Application - spring configuration for our app
+2. ItemController - handles the web traffic and interacts with the Service
+3. ItemRepository - defines key and value of our data model
+4. ItemService - uses a Repository and can utilize business logic to adapt GemFire data into a nicer format 
+
+
+The web application will retrieve a request from the user, and then ask GemFire for the data in it's `item` region. 
+For read operations to the region, if the data exists in the GemFire region, the data will be retrieved without using the CacheLoader. 
+If the data does **not** exist in the GemFire region, the data will be fetched via the CacheLoader interface defined above.
+
+For write operations to the GemFire region, the data is written to the GemFire region first and then batched up to be written to postgres using the AsyncEventListener.
+
+## Start the Spring Boot web service with GemFire integration
+
+Connecting a GemFire client to an existing GemFire cluster is easy. 
+In this example, we have a simple client that passes all reads an writes to the web server. 
+The web server is a Spring Boot web service configured to delegate all read and write operations
+to a GemFire cluster using Spring Boot and Spring Data.  
```

Comment:
Can we call out that it's using Spring for GemFire? Instead of Spring Boot and Spring Data.

Reply:
Specifically:
The web server is a Spring Boot web service configured to delegate all read and write operations to a GemFire cluster using Spring Boot for VMware GemFire.